File: admin_frontend/types/coworking.ts
export interface Coworking {
  id: number;
  name: string;
  description: string;
  address: string;
  workingHoursLabel: string;
  heroTitle?: string | null;
  heroText?: string | null;
  imageUrls: string[];
  uploadedImageUrls?: string[];
  uploadedImageFileIds?: string[];
  schedule: number;
  autoApproveMembership: boolean;
  floorMapEnabled: boolean;
  joinToken?: string | null;
  active: boolean;
  archived: boolean;
  configurationVersion: number;
  archivedAt?: string | null;
  createdAt?: string;
  updatedAt?: string;
}

export interface CreateCoworkingRequest {
  name: string;
  description: string;
  address: string;
  workingHoursLabel: string;
  heroTitle?: string;
  heroText?: string;
  autoApproveMembership?: boolean;
  floorMapEnabled?: boolean;
}

export interface UpdateCoworkingRequest {
  name: string;
  description: string;
  address: string;
  workingHoursLabel: string;
  heroTitle?: string;
  heroText?: string;
  imageFileIds?: string[];
  autoApproveMembership?: boolean;
  floorMapEnabled?: boolean;
  active?: boolean;
}

export interface CoworkingJoinLink {
  coworkingId: number;
  joinToken: string | null;
  joinUrl: string | null;
}

export interface CoworkingDashboard {
  coworking: Coworking;
  grants: string[];
  subjectLabel: string;
}

File: admin_service/src/main/java/com/hse/adminservice/coworking/domain/Coworking.java
package com.hse.adminservice.coworking.domain;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "coworkings")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Coworking {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, length = 2000)
    private String description;

    @Column(nullable = false, length = 500)
    private String address;

    @Column(nullable = false, length = 255)
    private String workingHoursLabel;

    @Column(length = 255)
    private String heroTitle;

    @Column(length = 2000)
    private String heroText;

    @Lob
    @Column(nullable = false)
    private String imageUrlsJson;

    @Column(nullable = false)
    private Integer schedule;

    @Column(nullable = false)
    private Long ownerId;

    @Column(name = "auto_approve_membership", nullable = false)
    private Boolean autoApproveMembership;

    @Column(name = "is_active", nullable = false)
    private Boolean active;

    @Column(nullable = false)
    private Boolean archived;

    @Column(nullable = false)
    private Long configurationVersion;

    private LocalDateTime archivedAt;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;
}
